### PR TITLE
Follows

### PR DESCRIPTION
**Pull request by Rynvasis**
add follows user with POST `/users/username/follow` and add get followers with GET `/users/username/followers` and add get following with GET `/users/username/following`

### CODE REVIEW

File: database.sql [hunk 1]
```diff
@@ -16,3 +16,10 @@ CREATE TABLE posts (
     content TEXT NOT NULL,
     createdAt TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
 );
+CREATE TABLE follows (
+    follower_id INT NOT NULL,
+    following_id INT NOT NULL,
+    FOREIGN KEY (follower_id) REFERENCES users(id),
+    FOREIGN KEY (following_id) REFERENCES users(id),
```
Comment (Salint) [inline database.sql:23]: Follow a consistent foreign key syntax. Use `REFERENCES`. Look at `posts` table for an example.

File: src/Services/UserService.cs [hunk 2]
```diff
@@ -57,5 +57,33 @@ public bool DoesUserExistById(int id)
...
+
+            using (var command = new MySqlCommand("INSERT IGNORE INTO follows (follower_id, following_id) VALUES (@followerId, @followingId)", DatabaseService.Connection))
+            {
+                command.Parameters.AddWithValue("@followerId", followerId);
+                command.Parameters.AddWithValue("@followingId", followingId);
+                int result = command.ExecuteNonQuery();
+
+                DatabaseService.CloseConnection();
+                return result > 0;
+            }
```
Comment (Salint) [inline src/Services/UserService.cs:73]: Use `try-finally` and put the connection closing in the `finally` block so if an error occurs, we make sure the connection is closed properly.

**Review by Salint — CHANGES_REQUESTED** [review]

File: src/Services/UserService.cs (hunk 2 above)
Comment (Rynvasis) [reply src/Services/UserService.cs:73]: okay, i will do that

File: database.sql (hunk 1 above)
Comment (Rynvasis) [reply database.sql:23]: okay

**Comment by Salint** [comment]
@Rynvasis 
Also, edit the "fetching user" endpoint to require authentication token and therefore you can use that to return "isFollowing" to the user profile so the frontend users can use it to determine if you're following the user or not.

Additionally, return the number of "followers" and "following".

**Pull request merged**